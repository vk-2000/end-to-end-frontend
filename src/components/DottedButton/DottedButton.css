.dotted-button {
    width: 100%;
    background-color: #e9eaf2;
    border: 2px dotted #5e40d7;
    border-radius: 10px;
    height: 50px;
    color: #5e40d7;
    font-size: 1rem;
    font-weight: 600;
    cursor: pointer;
}