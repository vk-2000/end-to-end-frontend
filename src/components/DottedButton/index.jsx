import React from 'react';
import PropTypes from 'prop-types';
import './DottedButton.css';

const DottedButton = ({ text, onClick }) => (
  <button className="dotted-button" onClick={onClick} type="button">
    {text}
  </button>
);

export default DottedButton;

DottedButton.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
