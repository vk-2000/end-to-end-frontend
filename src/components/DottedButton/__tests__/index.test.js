import { render } from '@testing-library/react';
import DottedButton from '..';

describe('DottedButton', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<DottedButton
      text="Hello"
      onClick={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should call onClick when clicke', () => {
    const onClick = jest.fn();
    const { getByText } = render(<DottedButton
      text="Hello"
      onClick={onClick}
    />);
    getByText('Hello').click();
    expect(onClick).toHaveBeenCalled();
  });
});
