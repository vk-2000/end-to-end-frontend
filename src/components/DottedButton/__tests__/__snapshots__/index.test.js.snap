// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DottedButton should render correctly 1`] = `
<DocumentFragment>
  <button
    class="dotted-button"
    type="button"
  >
    Hello
  </button>
</DocumentFragment>
`;
